diam/sm/smparser/cea.go

```diff
@@ -21,13 +21,16 @@ type CEA struct {
 	AcctApplicationID           []*diam.AVP               `avp:"Acct-Application-Id"`
 	AuthApplicationID           []*diam.AVP               `avp:"Auth-Application-Id"`
 	VendorSpecificApplicationID []*diam.AVP               `avp:"Vendor-Specific-Application-Id"`
+	FailedAVP                   []*diam.AVP               `avp:"Failed-AVP"`
+	ErrorMessage                string                    `avp:"Error-Message"`
 	appID                       []uint32                  // List of supported application IDs.
 }
 
 // ErrFailedResultCode is returned by Dial or DialTLS when the handshake
 // answer (CEA) contains a Result-Code AVP that is not success (2001).
 type ErrFailedResultCode struct {
 	Code uint32
+	Cea  *CEA
```

Comment:
We should call this field CEA (not Cea) and remove the Code field due to duplication. I wouldn't mind a `Code()` method in this struct as a shortcut to return `CEA.ResultCode`.